<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LifemapR">
<title>run-LifemapR • LifemapR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="run-LifemapR">
<meta property="og:description" content="LifemapR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LifemapR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/runLifemapR.html">run-LifemapR</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>run-LifemapR</h1>
                        <h4 data-toc-skip class="author">Cassandra
Bompard</h4>
            
            <h4 data-toc-skip class="date">2023-10-26</h4>
      
      
      <div class="d-none name"><code>runLifemapR.Rmd</code></div>
    </div>

    
    
<p>Introducing LifemapR, a package aiming to visualise data on a Lifemap
basemap. The uniqueness of LifemapR is that it offers interactive
visualization of data across a vast taxonomy (the NCBI taxonomy).
LifemapR can easily process large datasets, exceeding 300,000 rows, with
remarkable efficiency and precision. But what sets LifemapR apart is its
flexibility and adaptability, offering a wide spectrum of customization
options. This allows users to personalize their data visualization to
better suit their unique needs and preferences.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Install the development version of <code>LifemapR</code> from
GitHub</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"damiendevienne/LifemapR"</span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Once installed, the package can be loaded with :</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"LifemapR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="section level3">
<h3 id="general-case">General case<a class="anchor" aria-label="anchor" href="#general-case"></a>
</h3>
<p>The very first step to use this package is to have a dataframe
containing at least a <code>taxid</code> column containing NCBI format
TaxIds. This dataframe can also contain other data that you might want
to visualise.</p>
<pre><code><span><span class="co">##    taxid                  sci_name zoom       lat       lon</span></span>
<span><span class="co">## 6   2836           Bacillariophyta   13 -8.303813 -11.45440</span></span>
<span><span class="co">## 7   2849             Phaeodactylum   22 -8.350041 -11.49067</span></span>
<span><span class="co">## 8   2850 Phaeodactylum tricornutum   24 -8.349969 -11.49059</span></span>
<span><span class="co">## 9   2857                 Nitzschia   19 -8.342116 -11.49004</span></span>
<span><span class="co">## 10  2864               Dinophyceae   12 -6.949512 -12.25927</span></span></code></pre>
<p>You can then transform this dataframe into a format suitable for the
visualisation functions of the package with the
<code>build_Lifemap</code> function.<br></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"LifemapR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LM</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_Lifemap.html">build_Lifemap</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></code></pre></div>
<p>After that, you get a <code>lifemap</code> object that takes the form
of a list containing the following informations :</p>
<ul>
<li>
<code>df</code> : containing the augmented dataframe obtained after
the first step.</li>
<li>
<code>basemap</code> : which basemap was used to fetch the
data.</li>
</ul>
<p>You can then visualise your data with <code><a href="../reference/lifemap.html">lifemap()</a></code>
associated with one or more of the following functions :</p>
<ul>
<li>
<code><a href="../reference/lm_markers.html">lm_markers()</a></code>, to represent data as circles.</li>
<li>
<code><a href="../reference/lm_branches.html">lm_branches()</a></code>, to represent the data’s subtree.</li>
<li>
<code>lm_discret()</code>, to represent discret data as
piecharts.</li>
</ul>
<p>Each one of these three function adds a layer to the visualisation.
These layers are combined with the <code>+</code> symbol.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example with default representation</span></span>
<span></span>
<span><span class="co"># one layer</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># three different layers </span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/lm_branches.html">lm_branches</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>These function also allow the user to represent data by modifying
characteristics of representations as we’ll see later in the
examples.</p>
<p>The output is a shiny interface where the user can move and zoom
freely.</p>
<p>Please note that the following examples have been done in may 2023,
if you try them you may not have the same values due to database
update</p>
</div>
<div class="section level3">
<h3 id="analyse-of-kraken2-results">Analyse of Kraken2 results<a class="anchor" aria-label="anchor" href="#analyse-of-kraken2-results"></a>
</h3>
<p>This dataset is the result of a classification by Kraken (Derrick E.
Wood, J. Lu, 2019) on a metagenomic sample coming from a controlled set
of 12 known bacterial species (V. Sevim, J. Lee, 2019).</p>
<p>First of all, we load the data and transform it into the LifemapR
format.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kraken_res</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LM_kraken</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_Lifemap.html">build_Lifemap</a></span><span class="op">(</span>df <span class="op">=</span> <span class="va">kraken_res</span>, basemap <span class="op">=</span> <span class="st">"ncbi"</span><span class="op">)</span></span></code></pre></div>
<p>Then we can began to visualize our data.</p>
<p>We can for example represent the number of read that was assigned to
each TaxID by the color of the markers with the following command.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_kraken</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>var_fillColor <span class="op">=</span> <span class="st">"coverage_percent"</span>, fillColor <span class="op">=</span> <span class="st">"PiYG"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/kraken_base.png" alt="visualisation of kraken data" width="40%" height="20%"><p class="caption">
visualisation of kraken data
</p>
</div>
<p>With this representation, the markers are displayed if the if the
associated node is close enough.</p>
<p>It is possible to change the way markers are displayed with the
<code>display</code> argument.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># All the nodes that were requested by the user</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_kraken</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>var_fillColor <span class="op">=</span> <span class="st">"coverage_percent"</span>, fillColor <span class="op">=</span> <span class="st">"PiYG"</span>, display <span class="op">=</span> <span class="st">"requested"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Only the nodes that have no descendants</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_kraken</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>var_fillColor <span class="op">=</span> <span class="st">"coverage_percent"</span>, fillColor <span class="op">=</span> <span class="st">"PiYG"</span>, display <span class="op">=</span> <span class="st">"leaves"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
&lt;img src=“figures/kraken_requested.png” alt=“left : display
=”requested”, right : display = “leaves”” width=“40%” height=“20%”
/&gt;&lt;img src=“figures/kraken_leaves.png” alt=“left : display
=”requested”, right : display = “leaves”” width=“40%” height=“20%” /&gt;
<p class="caption">
left : display = “requested”, right : display = “leaves”
</p>
</div>
<p>Informations can also be displayed for the user with the
<code>popup</code> or <code>label</code> arguments.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># When clicking on a node, display the desired information</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_kraken</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>var_fillColor <span class="op">=</span> <span class="st">"coverage_percent"</span>, fillColor <span class="op">=</span> <span class="st">"PiYG"</span>, popup <span class="op">=</span> <span class="st">"name"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/kraken_popup.png" alt="Usage of the ```popup``` argument" width="40%" height="20%"><p class="caption">
Usage of the <code>popup</code> argument
</p>
</div>
<p>Finally, it is also possible to represent data with a subtree, either
by the size of the branches or their color.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Information on branche's color</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_kraken</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_branches.html">lm_branches</a></span><span class="op">(</span>var_color <span class="op">=</span> <span class="st">"coverage_percent"</span>, color <span class="op">=</span> <span class="st">"PiYG"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Information on branche's size</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_kraken</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_branches.html">lm_branches</a></span><span class="op">(</span>size <span class="op">=</span> <span class="st">"coverage_percent"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/kraken_branches_color.png" alt="left : branche's color, right : branche's size" width="45%" height="30%"><img src="figures/kraken_branches_size.png" alt="left : branche's color, right : branche's size" width="45%" height="30%"><p class="caption">
left : branche’s color, right : branche’s size
</p>
</div>
</div>
<div class="section level3">
<h3 id="comparative-genomic-data-on-genome-sizes-and-transposable-elements">comparative genomic data on genome sizes and transposable
elements<a class="anchor" aria-label="anchor" href="#comparative-genomic-data-on-genome-sizes-and-transposable-elements"></a>
</h3>
<p>This dataset contains informations about the genome size and the
Transposable Elements content for molluscs, insects and vertebrates.</p>
<p>First of all, we load the data and transform it into the LifemapR
format.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gen_res</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LM_gen</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_Lifemap.html">build_Lifemap</a></span><span class="op">(</span>df <span class="op">=</span> <span class="va">gen_res</span>, basemap <span class="op">=</span> <span class="st">"ncbi"</span><span class="op">)</span></span></code></pre></div>
<p>Then we can began to visualize our data.</p>
<p>Here we have two characteristics to visualise, we can do so with the
following command.</p>
<p>However, unlike the precedent data set, we don’t have informations
for all the nodes. Here we only have data for the leaves so it will be
necessary to infere values to the nodes where the information is missing
with the <code>FUN</code> argument.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualisation of the Genome size on the fillColor and the TEcontent on the size of markers.</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_gen</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>var_fillColor <span class="op">=</span> <span class="st">"Genome_size"</span>, fillColor <span class="op">=</span> <span class="st">"PiYG"</span>, radius  <span class="op">=</span> <span class="st">"TEcontent_bp"</span>, FUN <span class="op">=</span> <span class="va">mean</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/gen_base.png" alt="visualisation of genomics data" width="100%" height="20%"><p class="caption">
visualisation of genomics data
</p>
</div>
<p>We can also represent markers and subtree at the same time</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualisation of the Genome size on the fillColor and the TEcontent on the size of markers.</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_gen</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_branches.html">lm_branches</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>var_fillColor <span class="op">=</span> <span class="st">"Genome_size"</span>, fillColor <span class="op">=</span> <span class="st">"PiYG"</span>, radius  <span class="op">=</span> <span class="st">"TEcontent_bp"</span>, FUN <span class="op">=</span> <span class="va">mean</span><span class="op">)</span> </span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/gen_markers_branches.png" alt="visualisation of genomics data with markers and subtree" width="100%" height="20%"><p class="caption">
visualisation of genomics data with markers and subtree
</p>
</div>
</div>
<div class="section level3">
<h3 id="dataset-of-eukaryotes-from-ncbi-database">dataset of eukaryotes from NCBI database<a class="anchor" aria-label="anchor" href="#dataset-of-eukaryotes-from-ncbi-database"></a>
</h3>
<p>This dataset contains informations about around 1 000 eukaryotes
randomly fetched from the NCBI database.</p>
<p>First of all, we load the data and transform it into the LifemapR
format.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">eukaryotes_1000</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LM_eukaryotes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_Lifemap.html">build_Lifemap</a></span><span class="op">(</span>df <span class="op">=</span> <span class="va">eukaryotes_1000</span>, basemap <span class="op">=</span> <span class="st">"ncbi"</span><span class="op">)</span></span></code></pre></div>
<p>Then we can began to visualize our data.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualisation of eukaryotes data.</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_eukaryotes</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/eukaryotes_base.png" alt="Basic visualisation" width="50%" height="20%"><p class="caption">
Basic visualisation
</p>
</div>
<p>We can also choose to visualise only a part of our data. To do this,
we can either sort our data in advance or use the <code>data</code>
argument to do so.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualisation of Plants.</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_eukaryotes</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_markers.html">lm_markers</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">LM_eukaryotes</span><span class="op">$</span><span class="va">df</span><span class="op">[</span><span class="va">LM_eukaryotes</span><span class="op">$</span><span class="va">df</span><span class="op">$</span><span class="va">Group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="st">"Plants"</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/eukaryotes_data.png" alt="Visualisation of Plants" width="50%" height="20%"><p class="caption">
Visualisation of Plants
</p>
</div>
<p>Finally we can visualise discret variables with the
<code>lm_piechats()</code> function as following.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualisation of the maximum assembly level.</span></span>
<span><span class="fu"><a href="../reference/lifemap.html">lifemap</a></span><span class="op">(</span><span class="va">LM_eukaryotes</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="../reference/lm_piecharts.html">lm_piecharts</a></span><span class="op">(</span>param <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="figures/eukaryotes_piecharts.png" alt="Visualisation of the maximum assembly level" width="50%" height="20%"><p class="caption">
Visualisation of the maximum assembly level
</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Cassandra Bompard, Damien M. de Vienne, Aurélie Siberchicot.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
